packages/malloy/src/lang/parse-malloy.ts

```diff
@@ -786,11 +803,7 @@ export abstract class MalloyTranslation {
       line: startToken.line - 1,
       character: startToken.charPositionInLine,
     };
-    if (
-      this.parseStep.sourceInfo &&
-      stopToken.stopIndex != -1 &&
-      stopToken.stopIndex != startToken.startIndex
-    ) {
+    if (this.parseStep.sourceInfo && stopToken.stopIndex != -1) {
```

Comment:
i prefer to explicity not look for the stop index if i already know it is the same as the start

Reply:
Okay, I see what's happening now. There was a bug with `stopToken.stopIndex != startToken.startIndex` because it turns out `stopIndex` is the index of the last character, not the index after the last character. So single-character tokens were getting the wrong locations. I fixed the bug by looking up the end location in this case, but the correct fix is `stopToken.stopIndex >= startToken.startIndex` or `stopToken.stopIndex != startToken.startIndex - 1`, which is the case you're trying to avoid looking up the end location.

Reply:
```
if (
  this.parseStep.sourceInfo &&
  stopToken.stopIndex != -1 &&
  // If the `stopToken.stopIndex` is one before the `startToken.startIndex`, it means
  // it's length 0 and we don't need to look up the end.
  stopToken.stopIndex != startToken.startIndex - 1
) {
```

Reply:
I think the only zero length token might be the EOF token, and that won't be a startToken, and so I think we can live with no change here.